src/email/validators.ts

```diff
@@ -1,6 +1,26 @@
+import { getEmailDomain } from './formatters'
+
 export const isEmail = (arg: string): boolean => {
   if (!arg || typeof arg !== 'string') return false
 
   const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
   return regex.test(arg)
 }
+
+export const hasPublicDomain = (email: string): boolean => {
+  if (!isEmail(email)) return false
+  const publicEmailDomain = [
+    'bol.com',
```

Comment:
To reduce the code, you can remove the `.com` and later do:

const isPublic = publicEmailDomain.some(publicDomain => String(emailDomain).includes(`${publicDomain}.com`))